src/MongoDB.EntityFrameworkCore/Diagnostics/MongoLoggerExtensions.cs

```diff
@@ -109,4 +110,105 @@ private static MqlQueryEventDefinition LogExecutedMqlQuery(IDiagnosticsLogger lo
     }
 
     private const string LogExecutedMqlQueryString = "Executed MQL query{newLine}{collectionNamespace}.aggregate([{queryMql}])";
+
+
+    public static void RecommendedMinMaxRangeMissing(
+        this IDiagnosticsLogger<DbLoggerCategory.Model.Validation> diagnostics,
+        IProperty property)
+    {
+        var definition = LogRecommendedMinMaxRangeMissing(diagnostics);
+
+        if (diagnostics.ShouldLog(definition))
+        {
+            definition.Log(diagnostics, property.DeclaringType.DisplayName(), property.Name,
+                BsonTypeHelper.GetBsonType(property).ToString());
+            ;
+        }
+
+        if (diagnostics.NeedsEventData(definition, out var diagnosticSourceEnabled, out var simpleLogEnabled))
+        {
+            var eventData = new PropertyEventData(
+                definition,
+                (d, p) => ((EventDefinition<string, string, string>)d).GenerateMessage(
+                    ((PropertyEventData)p).Property.DeclaringType.DisplayName(),
+                    ((PropertyEventData)p).Property.Name,
+                    BsonTypeHelper.GetBsonType(((PropertyEventData)p).Property).ToString()),
+                property);
+            diagnostics.DispatchEventData(definition, eventData, diagnosticSourceEnabled, simpleLogEnabled);
+        }
+    }
+
+    private static EventDefinition<string, string, string> LogRecommendedMinMaxRangeMissing(IDiagnosticsLogger logger)
+    {
+        var definition = ((MongoLoggingDefinitions)logger.Definitions).LogRecommendedMinMaxRangeMissing;
+        if (definition == null)
+        {
+            definition = NonCapturingLazyInitializer.EnsureInitialized(
+                ref ((MongoLoggingDefinitions)logger.Definitions).LogRecommendedMinMaxRangeMissing,
+                logger,
+                static logger => new EventDefinition<string, string, string>(
+                    logger.Options,
+                    MongoEventId.RecommendedMinMaxRangeMissing,
+                    LogLevel.Warning,
+                    "MongoEventId.RecommendedMinMaxRangeMissing",
+                    level => LoggerMessage.Define<string, string, string>(
+                        level,
+                        MongoEventId.RecommendedMinMaxRangeMissing,
+                        LogRecommendedMinMaxRangeMissingString)));
+        }
+
+        return (EventDefinition<string, string, string>)definition;
+    }
+
+    private const string LogRecommendedMinMaxRangeMissingString =
+        "The property '{entityType}.{propertyType}' is configured for Queryable Encryption range queries but is missing the recommended min/max values required for {bsonType} storage.";
```

Comment:
If there is a good docs page to go to, then adding a fwd/go/whatever short URL here pointing to it might be useful. Really just something to consider any time we have a message.

Reply:
We write the docs after the feature so we could come back and update it later. Perhaps I should create a ticket to come back through the various custom exceptions we have and write a troubleshooting page that has them all on it?

Reply:
The value may not be there to come back and do it later. I'll make a note to bring it up with people. (Also, if the link it a a redirection not a direct link, then the link can be created now as a placeholder and updated to point to the docs when they are done. It can even point to the general server docs for queryable encryption until then. But anyway, more just something to think about.)